osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/application/services/DeviceManagementService.java

```diff
@@ -138,27 +140,28 @@ public void addEventNotifications(final String deviceIdentification,
         return ssldDevice.getOutputSettings();
     }
 
-    public void sendMeasurements(final String deviceIdentification, final GetDataResponseDto response)
-            throws ProtocolAdapterException {
+    public void sendMeasurements(final String deviceIdentification, final GetDataResponseDto response) {
         // Correlation ID is generated @ WS adapter, domain+version is
         // hard-coded
         // for now
-        final ProtocolResponseMessage responseMessage = new ProtocolResponseMessage.Builder().dataObject(response)
-                .deviceMessageMetadata(new DeviceMessageMetadata(deviceIdentification, "no-organisation",
-                        "no-correlationUid", DeviceFunctionDto.GET_DATA.name(), 0))
-                .result(ResponseMessageResultType.OK).domain("MICROGRIDS").domainVersion("1.0").build();
+        final ProtocolResponseMessage responseMessage = new ProtocolResponseMessage.Builder().dataObject(
+                response).deviceMessageMetadata(
+                new DeviceMessageMetadata(deviceIdentification, NO_ORGANISATION, NO_CORRELATION_UID,
+                        MessageType.GET_DATA.name(), 0)).result(ResponseMessageResultType.OK).domain(
+                "MICROGRIDS").domainVersion("1.0").build();
```

Comment:
Yet another puppy died here.

You could extract the hard-coded values as constants. @phoutman are there any ideas about future domains and versions of protocol messages?

Reply:
Note that this code is ONLY touched because I replaced a deprecated class. I did not look/evaluate the existing code because it is way out of scope of this ticket.